### DISCUSSION TITLE
Wordpress Interactivity API in Woocommerce's Cart Fragments architecture, following the 9.2 decision to update product stock at checkout only.

### DISCUSSION THREAD

**Discussion by oxfordmetadata** [woocommerce/woocommerce#50939] [Checkout Flow]
Hello, the wordpress team has produced this demo of their brand new Interactivity API

https://github.com/WordPress/wp-movies-demo?tab=readme-ov-file

https://wpmovies.dev/

Please notice the 0 delay interactions in the add to favourite action. 

For a long time Woocommerce's pain point is the delay of the "add to cart", "remove from cart" actions - especially in shops with 1000s of products and 10000s of customers. 

We have clients with complex catalogues that the add to/remove from cart action may take up to 3 seconds right now. In the particular shop (like a grocery store where the average item count is over 10 items this is a significant pain point). 

In the latest version of Woocommerce the team decided, and quite rightly so, to not update the stock quantity in the database upon the users add to cart/remove from cart actions: from now on stock updates will take place after the successful checkout. 

Given these two developments (i..e. the Interaction API and stock update at checkout) is it not the time to consider replacing the cart_fragments architecture and just move to an "add to cart" implementation using the Interactivity API? 

Following the latest change, it appears that there is absolutely NO POINT in having cart fragments server side. This will speed up our sites over all, as this has the biggest impact in our performance.

Many of our clients are truly concerned about the add to cart action delays - given the immediate interaction that other competitors (OpenCart, Magento, even NopCommerce) offer.

Thank you for your consideration. 

Dimitris Vayenas.  
Oxford Metadata, UK

**Comment by alexflorisca** [comment]
Thanks for your comment Dimitris. We are actively looking at ways to integrate the Interactivity API in WC at the moment. We are on the same page as you and believe it can simplify and speed up some parts of the codebase significantly. We are currently working on converting the Mini Cart to use interactivity API, but we will keep this in mind. However it's worth pointing out that the blocks version of the add to cart button uses the StoreApi, which I believe is faster than the cart_fragments implementation. And we are not actively developing the shortcode versions of store pages, so while we are considering the Interactivity API for developing our blocks, it is unlikely to get a lot of attention in the shortcode versions of the store pages/cart/checkout/etc.

**Reply by oxfordmetadata** [reply]
Thank you very much indeed @alexflorisca 

I hope that you will address the issue of the add to cart action (be it on a product listing page, or in a single product page). I do not know the particularities of where/when StoreApi or cart fragments are being used.  I am just stating the obvious - from the end user perspective.

When I am in any page; the quantities in the cart(s) ought to be "instantly" updated (both in the header and in the mini/side cart). 

[Here is a demo page:](https://femme-fatale.gr/product/makigiaz/cheili/lip-balm-lip-scrub/mua-hydra-juice-peptide-balm)

![image](https://github.com/user-attachments/assets/48418002-cce2-462d-91d1-32029bb997c9)

I hope that we are on the same page.

Right now I just do not know and cannot understand why the add to cart functionality ought to have a server side action, be it on a block, on a shortcode, or not. 

If you don't implement in the shortcode version (that many if not most theme developers are utilising) then the bottleneck will still be there, if anything it may make the frustration even more aparrent (if say the minicart get instantly updated but the number on the icons do not).
